Use the Wizard to Deploy with Catalog

katieperry4 opened the pull request:
<!--- If this is a non-code change, this template is not required; reference any issues or top-level descriptions as needed -->
<!--- All code change PRs should relate to an issue, reference it here; see example below -->
<!--- https://issues.redhat.com/browse/RHOAIENG-123456 -->
Closes [RHOAIENG-37348](https://issues.redhat.com/browse/RHOAIENG-37348)

## Description
<!--- Describe your changes in detail; the what, the why, any findings, etc -->
<!--- Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->
Allows you to deploy from catalog using the wizard
Starts you on the 2nd step of the wizard but you can still navigate back to the first step to see all the prefilled data
<img width="1001" height="670" alt="Screenshot 2025-11-06 at 9 57 15 AM" src="https://github.com/user-attachments/assets/bb0b2b3a-c094-4fbf-b363-ca309885868e" />
<img width="1005" height="693" alt="Screenshot 2025-11-06 at 9 57 29 AM" src="https://github.com/user-attachments/assets/5597f5b8-3ad6-419f-bf02-ff219c8c9b09" />
(also prefills the deployment name (same behavior as the modal) but you can edit that of course)

select a project and deploy as usual
after deploying it redirects to the global deployments page
if you hit cancel it takes you back to the catalog page

video: 

https://github.com/user-attachments/assets/e1d6fc46-56d8-48c2-8b55-7edd8f6d8a52



## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Tested locally

To test:
- Get catalog running locally (dm me if you need instructions on doing that)
- Go to catalog and pick a model
- hit the deploy button
- you should end up on the 2nd page of the wizard with the deployment name prefilled, and the project selector showing
- go back to the first step and see the location and model type prefilled and disabled
- finish deploying the model (pick a project, serving runtime, etc)
- get redirected to the global models page
- see the deployment
- edit the deployment see the location info prefilling as it should


## Test Impact
<!--- What tests have you done to cover the implemented functionality -->
<!--- If tests are not applicable, explain why here -->
No tests changed yet, all the model catalog mocked [tests](https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/__tests__/cypress/cypress/tests/mocked/modelCatalog/modelCatalogDeploy.cy.ts) are being skipped right now?

## Request review criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] The code follows our [Best Practices](/docs/best-practices.md) (React coding standards, PatternFly usage, performance considerations)

If you have UI changes: 
<!--- You can ignore these if you are doing manifest, backend, internal logic, etc changes; aka non-UI / visual changes -->
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Model Catalog can navigate to the deployment wizard with prefilled model data via a new navigation extension, hook, and wrapper; package export updated.
  * Deployment wizard supports an initial start index and a cancel return route.

* **Improvements**
  * Inputs across the deployment wizard and connection forms now support a disabled state.
  * Connection creation can hide entire sections for streamlined flows.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

katieperry4 commented:
/retest

katieperry4 commented:
@vconzola yeah mine doesn't have maas and I don't think I flipped all the feature flags for it to show up during that recording 👍

katieperry4 commented on packages/model-serving/src/components/deploymentWizard/ModelDeploymentWizard.tsx:
```diff
@@ -179,6 +179,7 @@ const ModelDeploymentWizard: React.FC<ModelDeploymentWizardProps> = ({
         onClose={() => setIsExitModalOpen(true)}
         onSave={() => onSave()}
         footer={wizardFooter}
+        startIndex={wizardState.initialData?.wizardStartIndex ?? 1}
```
no 🤣

katieperry4 commented on packages/model-serving/src/components/deploymentWizard/fields/ModelLocationInputFields.tsx:
```diff
@@ -56,7 +57,52 @@ export const useModelLocationData = (
+        });
+        setIsStableState(true);
+        setPrefillApplied(true);
+      }
```
if we enter the wizard without a project and there's existing data (data we pass in from catalog) with the type = NEW then we try and figure out the connection type from the uri we get passed in

I was only able to specify URI and S3 bc we can't use the whole OCI situation with catalog since we use it as a URI, but maybe since I moved the extension into our package I could move the setting of the connection type into there instead of here 🤔 I'll look at doing that

Griffin-Sullivan replied:
yeah that would be a good improvement

mturley commented on packages/model-registry/upstream/frontend/src/odh/components/ModelCatalogDeployModalExtension.tsx:
```diff
@@ -20,75 +25,66 @@ const ModelCatalogDeployModalExtension: React.FC<ModelCatalogDeployModalExtensio
   const params = useParams<CatalogModelDetailsParams>();
   const decodedParams = decodeParams(params);
-  const [artifacts, artifactLoaded, artifactsLoadError] = useCatalogModelArtifacts(
+  const [artifacts] = useCatalogModelArtifacts(
```
Removing `artifactLoaded` and `artifactsLoadError` here means we won't show a loading spinner or handle errors if the artifacts fail to load. I guess that's ok since the model details page where this is rendered will already handle that, but I wonder if what we should actually do is not bother fetching artifacts here, and instead pass the artifacts (or just the uri) down from the catalog page where we already have them loaded (ModelDetailsPage -> ModelCatalogDeployButton -> ModelCatalogDeployModalExtension).

Either way, we should probably have the deploy button be disabled if the artifacts are still loading or there was an error loading them.

mturley commented on packages/model-registry/package.json:
```diff
@@ -45,8 +45,10 @@
+    "./extensions": "./extensions.ts",
+    "./model-catalog-deploy": {
+      "types": "./upstream/frontend/src/odh/extension-points/model-catalog-deploy.ts"
+    }  },
```
Hm, we don't have linters on these package.json files I guess - the last `},` here should be on its own line

katieperry4 commented:
@mturley 
thanks for looking at this!
-I fixed the routing problem and now the navigate can handle separate submit and cancel routes
-Did a buncha renaming
-fixed the package.json line
-passed in a modelType, that defaults to generative if not provided
-put back the artifact loading and attached it to the button enablement state

I tried to reproduce the issue you were having but never saw it? What settings do you have in incognito mode?

https://github.com/user-attachments/assets/48572357-5130-4261-9636-a16a0a685dd2

it does sometimes take a hot second for the data to prefill on the model location step, that is something I need to look at 😅
